es-spatial-plugin/src/cmr/elasticsearch/plugins/spatial/script/core.clj

```diff
@@ -28,23 +30,64 @@
 ;;                         Begin script helper functions                     ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(def parameters
+  "The parameters to the Spatial script"
+  [:ords :ords-info :operator])
+
+(defn- extract-params
+  "Extracts the parameters from the params map given in the script."
+  [script-params]
+  (when script-params
+    (into {} (for [param parameters]
+               [param (XContentMapValues/nodeStringValue
+                        (get script-params (name param)) nil)]))))
+
 (defn- get-from-fields
   [^LeafStoredFieldsLookup lookup key]
   (when (and lookup key (.containsKey lookup key))
     (when-let [^FieldLookup field-lookup (.get lookup key)]
       (seq (.getValues field-lookup)))))
 
+(defn any-true?
+  "Returns true if predicate f returns a truthy value against any of the items.
+  This is very similar to some but it's faster through it's use of reduce."
+  [f items]
+  (reduce (fn [_ i]
+            (if (f i)
+              (reduced true) ;; short circuit
+              false))
+          false
+          items))
+
+(defn every-true?
+  "Returns true if predicate f returns a truthy value against all of the items."
+  [f items]
+  (every? true?
+    (map f items)))
+
+(defn remove-br
+  "Removes bounding rectangles from granule spatial data."
+  [shapes]
+  (let [shapes-no-br (into [] (remove nil? (for [shape shapes]
+                                             (if (not (= 5 (count shape)))
```

Comment:
Add comment on why `(= 5 (count shape))` means it is a bounding rectangle. Also there is no need to have the `for` loop.

Reply:
Changed logic per your suggestion and added clarifying comment for the use of `(= 5 (count ...))`

Reply:
Changed to use `(instance)` instead of `(count)`.